### ISSUE TITLE
Exporter reports Exim as down, even though service is running

### ISSUE THREAD

**Issue by Prime404** [gvengel/exim_exporter#10]
Hello,
I am facing issues with that the collector reports the Exim service as down, even though it is running.

Firstly this is the environment I am running running the collector using:
- CloudLinux release 8.6 (Leonid Kadenyuk)
- Exim version 4.95 #2 built 15-Jun-2022 12:59:12 (which exim /usr/sbin/exim)

If I run a basic check to test for the "exim_up" value I see the following:
```
#! curl -L localhost:9636/metrics | grep exim_up
exim_up 0
```

But if I check the status of the systemd unit I can see that the service is running:
```
● exim.service - Exim is a Mail Transport Agent, which is the program that moves mail from one machine to another.
   Loaded: loaded (/etc/systemd/system/exim.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2022-07-29 02:25:35 CEST; 6 days ago
 Main PID: 1963764 (exim)
    Tasks: 1 (limit: 298284)
   Memory: 16.0M
   CGroup: /system.slice/exim.service
           └─1963764 /usr/sbin/exim -ps -bd -q15m -oP /var/spool/exim/exim-daemon.pid
```

The exporter is running with the following parameters:
```
/opt/exim-exporter/bin/exim_exporter --exim.log-path="/var/log" \
--exim.mainlog="exim_mainlog" --exim.rejectlog="exim_rejectlog" \
 --exim.paniclog="exim_paniclog" --exim.executable="exim" \ 
--exim.input-path="/var/spool/exim/input"
```

My guess is that the exporter cannot find the correct pid-file, but there is no way to set the location of the PID-file in the application based on the env help list returned by the -h command.

**Comment by gvengel** [comment]
Thanks for the report, @Prime404!

The exporter doesn't actually use the PID file to detect whether the master daemon is running, since not all installations use one. Instead it looks for an exim process in daemon mode (via the `-bd` flag) which also is the leader of a process group. It's possible this method doesn't work on all systems and may need adjustment. 

Can you send me the output of `pstree -psg $(</var/spool/exim/exim-daemon.pid)` on the affected system?

Meanwhile I'll find some time to setup a system to reproduce the problem.

🍻

**Comment by gvengel** [comment]
Hmm, I setup a CloudLinux box and installed exim from the EPEL repository, and didn't run into any issue. Taking another look at your original message, I see you either wrote your own systemd unit or are using a package from somewhere else. Can you provide me with instructions on how to reproduce your setup? I probably just need a copy of `/etc/systemd/system/exim.service` and `/etc/systemd/system/exim_exporter.service` (or similar).

**Comment by Prime404** [comment]
Hello again,
The Exim version running is packaged by cPanel, below is the exact version running:
```
#! rpm -qa | grep exim
cpanel-exim-4.95-5.cp11104.x86_64
```

This packaged version of Exim was compiled with the following flags:
```
#! rpm -q --queryformat="%{NAME}: %{OPTFLAGS}\n" cpanel-exim-4.95-5.cp11104.x86_64
cpanel-exim: -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
```

**/etc/systemd/system/exim.service:**
```
[Unit]
ConditionPathExists=!/etc/eximdisable
Description=Exim is a Mail Transport Agent, which is the program that moves mail from one machine to another.
Wants=clamd.service spamd.service mailman.service

[Service]
Type=forking
EnvironmentFile=/etc/sysconfig/exim
PIDFile=/var/spool/exim/exim-daemon.pid
ExecStart=/usr/libexec/exim.daemon start
ExecReload=/usr/libexec/exim.daemon reload

[Install]
WantedBy=multi-user.target
```

And lastly the unit for the exporter:
```
[Unit]
Description=Exim Exporter
After=network.target

[Service]
User=root
Type=simple
ExecStart=/opt/exim-exporter/bin/exim_exporter --exim.log-path="/var/log" --exim.mainlog="exim_mainlog" --exim.rejectlog="exim_rejectlog" --exim.paniclog="exim_paniclog" --exim.executable="exim" --exim.input-path="/var/spool/exim/input"

[Install]
WantedBy=multi-user.target
```

As it might be useful the exim environment file is the following:
```
LISTENER_PID_FILE=/var/spool/exim/exim-daemon.pid
OUTGOING_PID_FILE=/var/spool/exim/exim-outgoing.pid
LANG=C
QUEUE=15m
TMPDIR=/tmp
DAEMON=yes
```

**Comment by gvengel** [comment]
Okay, this should be fixed in version 1.4.1. Let me know if you are still seeing issues.

**Comment by Prime404** [comment]
> Okay, this should be fixed in version 1.4.1. Let me know if you are still seeing issues.

Thanks for looking into this, I can confirm it works as it should now.  🍻
```
#! curl 127.0.0.01:9636/metrics -s | grep exim_up
# HELP exim_up Whether or not the main exim daemon is running
# TYPE exim_up gauge
exim_up 1
```

**Comment by shobhitrawat** [comment]
I am getting the same issue even though I am using the latest version of exim exporter 1.4.1.

I am running the docker image of Exim with the below configuration.

docker-compose.yml 

version: '3'
services:
  smtp:
    build: .
    image: smtp
    volumes:
       - /var/log/exim4:/var/log/exim4
       - /var/spool/exim4:/var/spool/exim4
    container_name: smtp
    restart: always
    ports:
      - "25:25"

Dockerfile

FROM ubuntu:18.04

RUN apt-get update && apt-get -y install telnet net-tools vim exim4 && chmod 777 -R /var/log/exim4 && chown -R root:root /var/log/exim4/

COPY    ./update-exim4.conf.conf        /etc/exim4/update-exim4.conf.conf
COPY    ./passwd.client		/etc/exim4/passwd.client
COPY	./exim4.conf.template	/etc/exim4/exim4.conf.template
COPY    ./02_exim4-config_options /etc/exim4/conf.d/main/02_exim4-config_options

EXPOSE 25
CMD ["/etc/init.d/exim4", "start"]

exim --version
Exim version 4.90_1 #4 built 30-Apr-2021 14:15:04
Copyright (c) University of Cambridge, 1995 - 2017

ps -ewF| grep exim
root          1      0  0  1159  1884  11 Aug26 ?        00:00:00 /bin/sh /etc/init.d/exim4 start
Debian-+    263      1  0 15388  6584  23 Aug26 ?        00:00:44 /usr/sbin/exim4 -bd -q30m

The exporter is running with the following parameters:

sudo docker run \
>   -p 9636:9636 \
>   -v /var/log/exim4:/var/log/exim4 \
>   -v /var/spool/exim4:/var/spool/exim4 \
>   -e exim.executable='exim4' \
>   --pid container:'smtp' \
>   --name exim_exporter \
>   gvengel/exim_exporter
ts=2022-08-30T05:48:56.268Z caller=main.go:360 level=info msg="Starting exim exporter" version="(version=1.4.1, branch=HEAD, revision=4b465da)"
ts=2022-08-30T05:48:56.268Z caller=main.go:361 level=info msg="Build context" context="(go=go1.19, user=, date=)"
ts=2022-08-30T05:48:56.268Z caller=main.go:250 level=info msg="Opening log" filename=/var/log/exim4/mainlog
ts=2022-08-30T05:48:56.268Z caller=main.go:250 level=info msg="Opening log" filename=/var/log/exim4/rejectlog
ts=2022-08-30T05:48:56.268Z caller=main.go:250 level=info msg="Opening log" filename=/var/log/exim4/paniclog
ts=2022-08-30T05:48:56.269Z caller=main.go:401 level=info msg=Listening address=:9636
ts=2022-08-30T05:48:56.269Z caller=tls_config.go:195 level=info msg="TLS is disabled." http2=false
ts=2022-08-30T05:48:56.269Z caller=log.go:198 ts="2022/08/30 05:48:56" caller="Seeked /var/log/exim4/mainlog - &{Offset:0" msg=" Whence:2}"
ts=2022-08-30T05:48:56.269Z caller=log.go:198 ts="2022/08/30 05:48:56" caller="Seeked /var/log/exim4/rejectlog - &{Offset:0" msg=" Whence:2}"
ts=2022-08-30T05:48:56.269Z caller=log.go:198 ts="2022/08/30 05:48:56" caller="Seeked /var/log/exim4/paniclog - &{Offset:0" msg=" Whence:2}"

 curl -L localhost:9636/metrics | grep exim_up
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP exim_up Whether or not the main exim daemon is running
# TYPE exim_up gauge
exim_up 0
100 18351    0 18351    0     0   943k      0 --:--:-- --:--:-- --:--:--  943k
